identity-actor/src/actor/endpoint.rs

```diff
@@ -0,0 +1,111 @@
+// Copyright 2020-2022 IOTA Stiftung
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::actor::Error;
+use crate::actor::Result as ActorResult;
+use std::fmt::Display;
+use std::str::FromStr;
+
+use serde::Deserialize;
+use serde::Serialize;
+
+// TODO: Deserialization skips endpoint validation, but that is okay-ish at present.
+/// A path-like identifier for a handler function. As an example, `identity/create`
+/// could be an endpoint of the "identity" namespace and the concrete action "create".
+///
+/// An endpoint can also represent a `hook` function, indicated by a `/hook` postfix.
+///
+/// Endpoints are separated by slashes and only allow alphabetic ascii characters and `_`.
+#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
+pub struct Endpoint {
+  name: String,
+  handler: String,
+  pub(crate) is_hook: bool,
+}
```

Comment:
As we discussed the `(name, handler)` split does currently not seem to be used after initialization and in many cases the endpoint will be known at compile time. Hence it might be useful to change the internal representation to the following: 
```suggestion 
#[derive(Debug, Clone, PartialEq, Eq, Hash, Serialize, Deserialize)]
pub struct Endpoint {
  name: Cow<'static, str>, 
  pub(crate) is_hook: bool
}
```

Reply:
Change the type to your suggestion, except without `is_hook`.